<launch>

  <!-- Box parameters TODO: NEEDS TUNING -->
  <param name="box_LX" type="double" value="0.045"/>
  <param name="box_LY" type="double" value="0.099"/>
  <param name="box_LZ" type="double" value="0.034"/>

  <!-- Nextage IK nodes -->
  <node pkg="rpbi_work" name="nextage_ik_node_left" type="nextage_ik_node.py" output="screen">
    <param name="arm" type="string" value="left"/>
  </node>

  <node pkg="rpbi_work" name="nextage_ik_node_right" type="nextage_ik_node.py" output="screen">
    <param name="arm" type="string" value="right"/>
  </node>

  <!-- get IK from box, TODO: check offset frame --> 
  <node pkg="rpbi_work" name="get_eff_pose_from_pushing_object" type="get_eff_pose_from_pushing_object.py" output="screen">
    <!-- <param name="offset_x" type="double" value="-0.02"/> -->
    <param name="offset_x" type="double" value="-0.012"/>
    <param name="offset_y" type="double" value="0.0"/>
    <!-- <param name="offset_z" type="double" value="0.02"/> -->
    <param name="offset_z" type="double" value="0.03"/>
  </node>

  <!-- Move nextage to pre push pose -->
  <node type="move_nextage_to_pre_push_pose.py" pkg="rpbi_work" name="move_nextage_to_pre_push_pose" output="screen"/>

  <!-- Remapper node -->
  <node name="remapper_node" pkg="rpbi_work" type="remap_joint_state_positions_to_float_array_node.py"/>

  <!-- Joao MPC node -->
  <!-- PyRBDL4dIK Environmet node (receiver - publisher) -->
  <node
      name="mpc_sliding_node_left"
      pkg="rpbi_work"
      type="nextage_ros_mpc_sliding_real_robot.py"
      required="true"
      output="screen">
    <param name="real_setup" value="true" type="bool"/>
    <param name="arm" value="left" type="string"/>
    <!-- List visual objects (optional) -->
    <rosparam param="object_config_file_name">
      [
      '{rpbi_work}/configs/nextage_sliding_box_visual.yaml',
      '{rpbi_work}/configs/nextage_sliding_box.yaml',
      ]
    </rosparam>
    <!-- Sys dyn param -->
    <rosparam param="sliding_param_dyn">
      [
      '{rpbi_work}/configs/sliding_dyn_param.yaml',
      ]
    </rosparam>
    <!-- Tracking to param -->
    <rosparam param="sliding_param_tracking_traj">
      [
      '{rpbi_work}/configs/nextage_tracking_config.yaml',
      ]
    </rosparam>
  </node>

  <!-- PyRBDL4dIK Environmet node (receiver - publisher) -->
  <node
      name="mpc_sliding_node_right"
      pkg="rpbi_work"
      type="nextage_ros_mpc_sliding_real_robot.py"
      required="true"
      output="screen">
    <param name="real_setup" value="true" type="bool"/>
    <param name="arm" value="right" type="string"/>
    <!-- List visual objects (optional) -->
    <rosparam param="object_config_file_name">
      [
      '{rpbi_work}/configs/nextage_sliding_box_visual.yaml',
      '{rpbi_work}/configs/nextage_sliding_box.yaml',
      ]
    </rosparam>
    <!-- Sys dyn param -->
    <rosparam param="sliding_param_dyn">
      [
      '{rpbi_work}/configs/sliding_dyn_param.yaml',
      ]
    </rosparam>
    <!-- Tracking to param -->
    <rosparam param="sliding_param_tracking_traj">
      [
      '{rpbi_work}/configs/nextage_tracking_config.yaml',
      ]
    </rosparam>
  </node>

  <!-- Move nextage to state node -->
  <node pkg="rpbi_work" name="move_nextage_to_state" type="move_nextage_to_state.py" output="screen"/>

  <!-- Node for filtering box-->
  <node pkg="rpbi_work" name="boxes_tf_filter" type="boxes_tf_filter.py" output="screen"/>



</launch>
